src/ensembl/src/content/app/browser/Browser.tsx

```diff
@@ -198,7 +200,14 @@ export const Browser: FunctionComponent<BrowserProps> = (
 
   useEffect(() => {
     const { activeGenomeId, fetchGenomeData } = props;
-    activeGenomeId && fetchGenomeData(activeGenomeId);
+    if (!activeGenomeId) {
+      return;
+    }
+    fetchGenomeData(activeGenomeId);
+    analyticsTracking.setCustomDimension({
+      diemension: CustomDimensions.SPECIES,
+      value: activeGenomeId
+    });
```

Comment:
I don't think this belongs in the Browser component. If what this code is saying is "set species dimension every time active genome id changes", it belongs either in browser actions, or, better yet, in a redux middleware.